lib/provider.js

```diff
@@ -31,27 +49,30 @@ module.exports = {
       return this.scanFeaturesDir(results)
     }
   },
-  scanStepDefinitionsDir: function(results) {
+  scanStepDefinitionsDir: function(results = []) {
     //TODO: first search step definitions for your file
-    for (var step_def_file of fs.readdirSync(`${this.rootDirectory()}${this.featuresDirectory()}/step_definitions`)) {
+    const searchPath = `${this.rootDirectory()}${this.featuresDirectory()}/step_definitions`;
+    const allStepPaths = this.searchForPattern(searchPath, /.*/);
+    allStepPaths.forEach( step_def_file => {
       let data = fs.readFileSync(`${this.rootDirectory()}${this.featuresDirectory()}/step_definitions/${step_def_file}`, 'utf8');
```

Comment:
Thanks for the PR update @Ellisande, looks good! I've loaded this up on my machine but it seems that `scanStepDefinitionsDir` is not working as expected and so the package is reverting to `scanFeaturesDir`. I think it's this line that isn't working correctly, maybe a result of the method refactor